init.sh

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env bash
+
+set -e
+
+usage() {
+  cat <<EOF
+NAME
+  init.sh - Initialize the project
+
+SYNOPSIS
+  init.sh [OPTION]
+
+DESCRIPTION
+  Initializes the project for a specific provider. Inputs can be provided as command line arguments or
+  will be prompted for if omitted and in an interactive shell.
+
+OPTIONS
+  -h, --help
+    Print this help message and exit.
+
+  -n <provider>
+    The provider name to initialize the project for (e.g. aws).
+
+  -r <repository>
+    The repository name to initialize the project for (defaults to "pulumi-\${provider}).
+
+  -o <organization>
+    The organization name to initialize the project for. This is typically the
+    GitHub username or organization name where the repository will live.
+
+  -u <upstream>
+    The upstream tf provider Go module to bridge into a Pulumi provider.
+    (e.g. hashicorp/terraform-provider-aws).
+
+  -d
+    Delete remaining boilerplate files after successful initialization
+
+  -v
+    Enable verbose mode. Prints each command executed.
```

Comment:
I think we should eventually aim to move this into something like `pulumi new provider`. It's very similar to a lot of the templating, renaming, etc. we already do for templates, and it should make this whole process much more discoverable to users.

Reply:
Sure, I think adding that as an option would be a good future improvement